test/anttasks/org/apache/cassandra/anttasks/TestNameCheckTask.java

```diff
@@ -0,0 +1,85 @@
+/*
```

Comment:
opinion: is `test/anttasks/` really the most intuitive place? I would have thought either `src/anttasks/` or `.build/anttasks/` would make more sense.

Reply:
given this is for tests I believe it is most suitable to put it among test sources, yes.

Reply:
i was thinking about all the anttasks classes. they are not part of the tests classpath, or the runtime classpath.

Reply:
@michaelsembwever the check implementation uses `org.junit.Test` annotation to scan test methods. org.junit.Test is on test class path and that annotation is not part of runtime classpath.

Reply:
> We would have to have special class path just for ant tasks with test classpath as well?

No. ant tasks are already their own compilation path with its own classpath. Thinking about it as runtime or test is wrong, it's more bootstrap than anything else. (It doesn't matter that the class itself uses org.junit.Test)

> Also I think that we need to have access to these test classes to scan etc so test class path is a must, no?

No. That's just runtime configuration of the ant task. (The ant task class doesn't need that classpath to compile.)

Moving all of the anttask folder to a new location is out-of-scope for this PR. But I still think it is the right thing to do (at a later point).

Reply:
Honestly I am not sure how this will look like but it is noted and we may create new ticket to move all ant tasks out of `test`. As of now, based on this communication, I consider this to be resolved in regards to this PR.